### ISSUE TITLE
Getting [DEPRECATED] Terraform environments are now called workspaces. Please use the workspace key instead.

### ISSUE THREAD

**Issue by Constantin07** [hashicorp/terraform#17814]
Hi,

I'm not using environments at all.
I'm using workspaces but still getting these misleading warnings when I ran `terraform plan`:

```shell
Warning: data.terraform_remote_state.base: "environment": [DEPRECATED] Terraform environments are now called workspaces. Please use the workspace key instead.

Warning: module.shared.data.terraform_remote_state.common: "environment": [DEPRECATED] Terraform environments are now called workspaces. Please use the workspace key instead.
```

### Terraform Version
```
terraform --version
Terraform v0.11.6
```

### Terraform Configuration Files
I'm using a 2 remote states in my terraform template which read some common output values defined in different workspaces.

### Expected Behavior
It should not print any warning about environments as I'm not using them.

### Actual Behavior
Printing misleading messages.

### Steps to Reproduce
1. `terraform init`
2. `terraform plan`
3. check the output.

**Comment by jbardin** [comment]
Hi @Constantin07,

That warning is coming from the remote state configuration, not the actual use of workspaces.
Do you have an `environment` attribute at all in the `terraform_remote_state` config?

**Comment by Constantin07** [comment]
@jbardin yes, something like that:

```hcl
data "terraform_remote_state" "base" {
  backend = "s3"

  environment = "${terraform.workspace}"

  config {
    bucket  = "base.state.terraform"
    region  = "eu-west-2"
    key     = "base.tfstate"
    encrypt = true
  }
}
```

In official documentation https://www.terraform.io/docs/providers/terraform/d/remote_state.html, the `environment` is still a valid option for remote state though.

Is there a way to get rid of these warnings ?

**Comment by jbardin** [comment]
The `environment` key is still valid, but deprecated, which is why you see the warnings.
Following the directions in the warning message to use `workspace` will prevent the warnings.

**Comment by Constantin07** [comment]
Thanks @jbardin, unfortunately changing `environment` to `workspace` doesn't help - I still see the warnings as before.

```hcl
data "terraform_remote_state" "base" {
  backend = "s3"

  workspace = "${terraform.workspace}"

  config {
    bucket  = "base.state.terraform"
    region  = "eu-west-2"
    key     = "base.tfstate"
    encrypt = true
  }
}
```

**Comment by jbardin** [comment]
Hi @Constantin07,

Did you remove `environment` from both locations?
Do you have any other modules which may be using `environment`?

**Comment by Constantin07** [comment]
@jbardin you are right. I found in module in a file with another remote state reference which had `environment`.
After replacing I don't get any warning anymore.

Many Thanks @jbardin for your help.
